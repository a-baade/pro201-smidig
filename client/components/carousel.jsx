import React, { useEffect, useState } from 'react'

export function CarouselCard(){
    return(
        <div className={"carousel-card"}>
            <div className={"card-header"}></div>
            <img src="https://dummyimage.com/300.png/09f/fff" alt="placeholder" className={"pfp-placeholder"}/>
            <p className={"carousel-card-text"}>Thank you for your generous gift to us! We are thrilled to have your support.
            <br/> <br/>You truly make the difference for us, and we are extremely grateful.</p>
            <div className={"cc-btn-wrap"}>
                <button className={"cc-btn"}>info</button>
                <button className={"cc-btn"}>cancel</button>
            </div>
        </div>
    );
}

const Carousel = (props) => {
    const {children, show} = props

    const [currentIndex, setCurrentIndex] = useState(0)
    const [length, setLength] = useState(children.length)

    useEffect(() => {
        setLength(children.length)
    }, [children])

    const next = () => {
        if (currentIndex < (length - show)) {
            setCurrentIndex(prevState => prevState + 1)
        }
    }

    const prev = () => {
        if (currentIndex > 0) {
            setCurrentIndex(prevState => prevState - 1)
        }
    }

    return (
        <div className="carousel-container">
            <div className="carousel-wrapper">
                {
                    currentIndex > 0 &&
                    <button onClick={prev} className="left-arrow">
                        &lt;
                    </button>
                }
                <div className="carousel-content-wrapper">
                    <div
                        className={`carousel-content show-${show}`}
                        style={{ transform: `translateX(-${currentIndex * (100 / show)}%)` }}
                    >
                        {children}
                    </div>
                </div>

                {
                    currentIndex < (length - show) &&
                    <button onClick={next} className="right-arrow">
                        &gt;
                    </button>
                }
            </div>
        </div>
    )
}

export default Carousel