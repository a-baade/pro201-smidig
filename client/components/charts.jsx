import { Chart } from "react-google-charts";

export const data1 = [
    ["Organization", "Donations"],
    ["Norgesgruppen", 1500000],
    ["DNB", 1225678],
    ["Varner", 740000],
    ["Komplett", 550000],
];

export const options1 = {
    title: "Donations from similar sized organizations",
    titleTextStyle: {color: "#ffffff"},
    chartArea: {
        width: "50%",
        backgroundColor: "#66a2ad"
    },
    backgroundColor: "#66a2ad",
    colors: ["#35656f"],
    hAxis: {
        title: "Donated",
        titleTextStyle: {color: "#ffffff"},
        minValue: 0,
    },
    vAxis: {
        title: "Organization",
        titleTextStyle: {color: "#ffffff"},
    },
};

export function Chart1(){
    return(
        <Chart
            className={"dashboard-charts"}
            id={"dashboard-chart1"}
            chartType="BarChart"
            data={data1}
            width="700px"
            height="500px"
            legendToggle
            options={options1}
        />
    );
}

export const data2 = [
    ["Month", "Donations"],
    ["January", 19000],
    ["February", 20000],
    ["March", 23000],
    ["April", 21000],
    ["May", 22000],
];

export const options2 = {
    title: "My donations the last year in USD",
    curveType: "function",
    colors: ["#35656f"],
    legend: { position: "bottom" },
};

export function Chart2(){
    return(
        <Chart
            className={"dashboard-charts"}
            id={"dashboard-chart2"}
            chartType="LineChart"
            width="700px"
            height="500px"
            data={data2}
            options={options2}
        />
    );
}

export default function charts(){

}