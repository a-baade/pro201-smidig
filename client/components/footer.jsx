import Facebook from "../pics/icons/facebook.png";
import Instagram from "../pics/icons/instagram.png";
import Snapchat from "../pics/icons/snapchat.png";
import Tiktok from "../pics/icons/tiktok.png";
import { Link } from "react-router-dom";
import { useState } from "react";
import React from "react";

export default function Footer() {
  const [email, setEmail] = useState("");

  return (
    <footer>
      <div className={"footerLinks"}>
        <Link to={"/"} className={"footerLink"}>
          Social Media
        </Link>
        <p>|</p>
        <Link to={"/"} className={"footerLink"}>
          Donate
        </Link>
        <p>|</p>
        <Link to={"/about"} className={"footerLink"}>
          About us
        </Link>
        <p>|</p>
        <Link to={"/contact"} className={"footerLink"}>
          Contact
        </Link>
      </div>

      <div className={"footerSocial"}>
        <a href={"https://www.facebook.com/"}>
          <img src={Facebook} alt={"facebook"} />{" "}
        </a>
        <a href={"https://www.instagram.com/"}>
          <img src={Instagram} alt={"insta"} />
        </a>
        <a href={"https://www.snapchat.com/"}>
          <img src={Snapchat} alt={"snap"} />{" "}
        </a>
        <a href={"https://www.tiktok.com/"}>
          <img src={Tiktok} alt={"tiktok"} />
        </a>
      </div>

      <div className={" footerCopyright "}>
        <p>Copyright Â© 2022 Melioria Impact. All rights reserved</p>
      </div>

      <div className={"footerSubscribe"}>
        <p>Subscribe here for news about our causes</p>
        <form>
          <input
            type="email"
            placeholder="example@email.com"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
          <button>Subscribe</button>
        </form>
      </div>
    </footer>
  );
}
