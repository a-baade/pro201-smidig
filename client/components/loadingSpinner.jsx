import React from "react";

export default function LoadingWheel() {
    return (
        <div>
            <div className="loading">
                <div></div>
                <div></div>
                <div></div>
                <div></div>
                <div></div>
                <div></div>
                <div></div>
                <div></div>
            </div>
        </div>
    )
}