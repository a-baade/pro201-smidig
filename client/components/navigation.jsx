import Search from "../pics/icons/search.png";import Profile from "../pics/icons/avatar.png";import Logo from "../pics/icons/logo.png"import { Link } from "react-router-dom";/*        Bytt ut p med Link når routing er på plass */export default function Navigation() {  return(    <nav>      <div>        <Link to={"/"}><img src={ Logo } className={ "logo" }/></Link>        <Link to={"/charities"}  className={ "navLinks" }>Charities</Link>        <p className={ "navLinks" }>About us</p>        <p className={ "navLinks" }>Contact</p>        <p className={ "navLinks" }>Other</p>      </div>      <div>        <img src={ Search } className={ "navIcons" }/>        <div className={ "profileDropdown" }>          <img src={ Profile } className={ "navIcons" }/>          <div className={ "dropdownContent" }>            <ul><Link to={"/login"} className={ "dropdownLink" }>Log in</Link></ul>            <ul><Link to={"/profile"} className={"dropdownLink"}>Profile</Link></ul>            <ul><p className={ "dropdownLink" }>My dashboard</p></ul>            <ul><Link to={"/login/endsession"} className={"dropdownLink"}>Log out</Link></ul>          </div>        </div>      </div>    </nav>  );}