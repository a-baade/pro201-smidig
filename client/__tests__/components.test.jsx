import React from "react";
import ReactDOM from "react-dom";
import { Link, MemoryRouter } from "react-router-dom";
import Navigation from "../components/navigation";
import Footer from "../components/footer";
import { Chart1, Chart2 } from "../components/charts";

describe("Navigation and Footer components", () => {
  it("should test navigation", function () {
    const divEl = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
      divEl
    );
    expect(divEl.innerHTML).toMatchSnapshot();
  });

  it("should test Link", function () {
    const link = document.createElement("Link");
    ReactDOM.render(
      <MemoryRouter>
        <Link to={"/about"} className={"navLinks"}>
          About us
        </Link>
      </MemoryRouter>,
      link
    );
    expect(link.innerHTML).toMatchSnapshot();
  });

  it("should test div element of footer", function () {
    const divEl = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>,
      divEl
    );
    expect(divEl.innerHTML).toMatchSnapshot();
  });

  it("should test footer element", function () {
    const footerEl = document.createElement("footer");
    ReactDOM.render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>,
      footerEl
    );
    expect(footerEl.innerHTML).toMatchSnapshot();
  });

  it("should test chart1", function () {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Chart1 />
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchSnapshot();
  });

  it("should test chart2", function () {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Chart2 />
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchSnapshot();
  });
});
