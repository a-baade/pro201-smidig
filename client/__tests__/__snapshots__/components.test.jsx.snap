// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation and Footer components should test Link 1`] = `"<a class=\\"navLinks\\" href=\\"/about\\">About us</a>"`;

exports[`Navigation and Footer components should test chart1 1`] = `""`;

exports[`Navigation and Footer components should test chart2 1`] = `""`;

exports[`Navigation and Footer components should test div element of footer 1`] = `"<footer><div class=\\"footer-links\\"><a class=\\"footer-link\\" href=\\"/\\">Social Media</a><p>|</p><a class=\\"footer-link\\" href=\\"/charities\\">Donate</a><p>|</p><a class=\\"footer-link\\" href=\\"/about\\">About us</a><p>|</p><a class=\\"footer-link\\" href=\\"/contact\\">Contact</a></div><div class=\\"footer-social\\"><a href=\\"https://www.facebook.com/\\"><img src=\\"[object Object]\\" alt=\\"facebook\\"></a><a href=\\"https://www.instagram.com/\\"><img src=\\"[object Object]\\" alt=\\"insta\\"></a><a href=\\"https://www.linkedin.com/\\"><img src=\\"[object Object]\\" alt=\\"linkedin\\"></a><a href=\\"https://www.snapchat.com/\\"><img src=\\"[object Object]\\" alt=\\"snap\\"></a></div><div class=\\"footer-copyright\\"><p>Copyright © 2022 Melioria Impact. All rights reserved</p></div><div class=\\"footer-subscribe\\"><p>Subscribe here for news about our causes</p><form><input type=\\"email\\" placeholder=\\"example@email.com\\" required=\\"\\"><button>Subscribe</button></form></div></footer>"`;

exports[`Navigation and Footer components should test footer element 1`] = `"<footer><div class=\\"footer-links\\"><a class=\\"footer-link\\" href=\\"/\\">Social Media</a><p>|</p><a class=\\"footer-link\\" href=\\"/charities\\">Donate</a><p>|</p><a class=\\"footer-link\\" href=\\"/about\\">About us</a><p>|</p><a class=\\"footer-link\\" href=\\"/contact\\">Contact</a></div><div class=\\"footer-social\\"><a href=\\"https://www.facebook.com/\\"><img src=\\"[object Object]\\" alt=\\"facebook\\"></a><a href=\\"https://www.instagram.com/\\"><img src=\\"[object Object]\\" alt=\\"insta\\"></a><a href=\\"https://www.linkedin.com/\\"><img src=\\"[object Object]\\" alt=\\"linkedin\\"></a><a href=\\"https://www.snapchat.com/\\"><img src=\\"[object Object]\\" alt=\\"snap\\"></a></div><div class=\\"footer-copyright\\"><p>Copyright © 2022 Melioria Impact. All rights reserved</p></div><div class=\\"footer-subscribe\\"><p>Subscribe here for news about our causes</p><form><input type=\\"email\\" placeholder=\\"example@email.com\\" required=\\"\\"><button>Subscribe</button></form></div></footer>"`;

exports[`Navigation and Footer components should test navigation 1`] = `"<nav><div><a href=\\"/\\"><img src=\\"[object Object]\\" class=\\"logo\\"></a><a class=\\"navLinks\\" href=\\"/charities\\">Charities</a><a class=\\"navLinks\\" href=\\"/about\\">About us</a><a class=\\"navLinks\\" href=\\"/contact\\">Contact</a></div><div><div class=\\"search-box\\"><input class=\\"search-text\\" type=\\"text\\" placeholder=\\"Search Anything\\"><a href=\\"#\\" class=\\"search-button\\"><img src=\\"[object Object]\\" class=\\"search\\"></a></div><div class=\\"profileDropdown\\"><img src=\\"[object Object]\\" class=\\"navIcons\\"><div class=\\"dropdownContent\\"><a class=\\"dropdownLink\\" href=\\"/login\\">Login</a></div></div></div></nav>"`;
